src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/integration/ReactiveCourseRepositoryIT.java

```diff
@@ -0,0 +1,197 @@
+
+    @Test
+    public void testFindByIdAsPublisher() {
+        final Mono<Course> byId = repository.findById(Mono.just(COURSE_ID_1));
+        StepVerifier.create(byId).expectNext(COURSE_1).verifyComplete();
+    }
+
+    @Test
+    public void testFindAllWithSort() {
+        final Flux<Course> sortAll = repository.findAll(Sort.by(Sort.Order.desc("name")));
+        StepVerifier.create(sortAll).expectNext(COURSE_4, COURSE_3, COURSE_2, COURSE_1).verifyComplete();
+    }
+
+    @Test
+    public void testFindByIdNotFound() {
+        final Mono<Course> idMono = repository.findById("10");
+        //  Expect an empty mono as return value
+        StepVerifier.create(idMono).expectComplete().verify();
+    }
+
+    @Test
+    public void testFindByIdAndPartitionKeyNotFound() {
+        final Mono<Course> idMono = repository.findById("10",
+            new PartitionKey(entityInformation.getPartitionKeyFieldValue(COURSE_1)));
+        //  Expect an empty mono as return value
+        StepVerifier.create(idMono).expectComplete().verify();
+    }
+
+    @Test
+    public void testFindAll() {
+        final Flux<Course> allFlux = repository.findAll();
+        StepVerifier.create(allFlux).expectNextCount(4).verifyComplete();
+    }
+
+    @Test
+    public void testInsert() {
+        final Mono<Course> save = repository.save(COURSE_5);
+        StepVerifier.create(save).expectNext(COURSE_5).verifyComplete();
+    }
+
+    @Test
+    public void testUpsert() {
+        Mono<Course> save = repository.save(COURSE_1);
+        StepVerifier.create(save).expectNext(COURSE_1).expectComplete().verify();
+
+        save = repository.save(COURSE_1);
+        StepVerifier.create(save).expectNext(COURSE_1).expectComplete().verify();
+    }
+
+    @Test
+    public void testDeleteByIdWithoutPartitionKey() {
+        final Mono<Void> deleteMono = repository.deleteById(COURSE_1.getCourseId());
+        StepVerifier.create(deleteMono).expectError(CosmosDBAccessException.class).verify();
+    }
+
+    @Test
+    public void testDeleteByIdAndPartitionKey() {
+        final Mono<Void> deleteMono = repository.deleteById(COURSE_1.getCourseId(),
+            new PartitionKey(entityInformation.getPartitionKeyFieldValue(COURSE_1)));
+        StepVerifier.create(deleteMono).verifyComplete();
+
+        final Mono<Course> byId = repository.findById(COURSE_ID_1,
+            new PartitionKey(entityInformation.getPartitionKeyFieldValue(COURSE_1)));
+        //  Expect an empty mono as return value
+        StepVerifier.create(byId).verifyComplete();
+    }
+
+    @Test
+    public void testDeleteByEntity() {
+        final Mono<Void> deleteMono = repository.delete(COURSE_4);
+        StepVerifier.create(deleteMono).verifyComplete();
+
+        final Mono<Course> byId = repository.findById(COURSE_ID_4);
+        //  Expect an empty mono as return value
+        StepVerifier.create(byId).expectComplete().verify();
+    }
+
+    @Test
+    public void testDeleteByIdNotFound() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/Microsoft/spring-data-cosmosdb/pullRequest?prid=4741022)